node/network/availability-distribution/src/requester/mod.rs

```diff
@@ -133,8 +148,15 @@ impl Requester {
 			Self::LEAF_ANCESTRY_LEN_WITHIN_SESSION,
 		)
 		.await?;
+		span.add_uint_tag("ancestors-in-session", ancestors_in_session.len() as u64);
+
 		// Also spawn or bump tasks for candidates in ancestry in the same session.
 		for hash in std::iter::once(leaf).chain(ancestors_in_session) {
+			let span = span
```

Comment:
I don't think we need this kind of verbosity.

Reply:
If we don't include a span here, how do you suggest we pass a span to `add_cores()` in the loop? Without passing a span to `add_cores()` there is no way to trace the execution of `FetchTask`.

Reply:
I see what you mean. In that case, the span name sounds misleading.

Reply:
I see that add_cores also creates an additional span. Maybe we can create the span there and just pass it a ref to the `request-chunks` span we created above.

Reply:
Will take a look now :)

Reply:
Agreed - naming was off. I renamed this span and the one above as `request-chunks-new-head` and `request-chunks-ancestor` accordingly

Reply:
Regarding `add_cores` also creating an additional span;

1. The `add_cores` call is invoked per ancestor hash. If I understood it correctly, we go [`SESSION_ANCESTRY_LEN`] back when requesting chunks, so we would need to associate a `request-chunk-ancestor` span with each ancestor.
2. Inside `add_cores`, there is a loop over all the occupied cores, and for some of the occupied cores the entries are vacant. If they are vacant, the new `FetchTask` is crated.

This way we have a relationship like this:
```
request-chunk-new-head
|
|__ request-chunk-ancestor (ancestor_0)
    |
    |___ check-fetch-candidate (occupied_core_0)
    |___ check-fetch-candidate (occupied_core_1)
         |___ fetch-task-config (when vacant)
              |__ run-fetch-chunk-task
                  |__ ++ 
    |___ check-fetch-candidate (occupied_core_N)
|
|_ |__ request-chunk-ancestor (ancestor_1)
        |___ check-fetch-candidate (occupied_core_0)
        |___ check-fetch-candidate (occupied_core_1)
        |___ check-fetch-candidate (occupied_core_N)
```

Reply:
Got it! Thanks, it will be interesting to see anything relate to what is slowing down PoV recovery.

Reply:
Definitely!